Fixes borg shaker constantly melting ice

SpookyTheFox opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Greetings from downstream land! For a while now, I and others have noticed that the service cyborg spams chemical reaction sounds and melts all the ice it makes. This PR attempts to fix that. Following in the footsteps of https://github.com/tgstation/tgstation/pull/57144 , this sets the temperature of all the reagents the borg shaker makes to 274.5K, a magic happy middle point which keeps the water liquid and the ice solid.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Fixes an oversight that spammed chemical reaction noises, cluttered the chat box, and made several drinks impossible.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: fixed the cyborg shaker
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

SpookyTheFox committed:
Update borghydro.dm

Added variable to set the temperature of generated reagents. This was copied from https://github.com/tgstation/tgstation/pull/57144

SpookyTheFox commented:
First time contributing to TG, I hope I did everything right!

I tested this locally both for intended functionality with the borg shaker, and also tested the medical borg hypo to make sure it didn't break, since it's closely related code wise.

MrDoomBringer commented:
but the point of a shaker is to slightly dilute the drink by melting the ice smh my realism

Mothblocks commented on code/modules/reagents/reagent_containers/borghydro.dm:
```diff
@@ -24,6 +24,7 @@ Borg Hypospray
 	var/charge_cost = 50
 	var/charge_timer = 0
 	var/recharge_time = 10 //Time it takes for shots to recharge (in seconds)
+	var/dispensed_temperature = DEFAULT_REAGENT_TEMPERATURE
```
This needs to have a doc comment above it.

https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md

SpookyTheFox replied:
If I understand this right, I should put 
`///Optional variable to override the temperature add_reagent() will use`
Is that right?

Mothblocks replied:
Yeah.

Mothblocks commented on code/modules/reagents/reagent_containers/borghydro.dm:
```diff
@@ -205,6 +206,7 @@ Borg Shaker
 	charge_cost = 20 //Lots of reagents all regenerating at once, so the charge cost is lower. They also regenerate faster.
 	recharge_time = 3
 	accepts_reagent_upgrades = FALSE
+	dispensed_temperature = 274.5
```
Where does this number come from? Ideally, this would either be dynamically created or derived from some define (like T0C).

SpookyTheFox replied:
The value is what I came up with after looking at /datum/chemical_reaction/ice and /datum/chemical_reaction/water, both of which are in others.dm and start at line 733. The required_temp for ice is 274, and the required_temp for water is 275. By choosing a temperature between the two, neither reaction occurs. I have tested this and confirmed that both ice and water, as well as any other reagents from the borg shaker will not perform the ice or water reaction when poured into a container.
As for using a define- In #57114, which I heavily referenced, they used "T0C + 0.85" which gives us exactly 274. I tried this, it was freezing the water as it was being produced by the shaker. I could use "T0C + 1.35" which would solve to 274.5K as well, but I thought just using 274.5 in the first place was better. I can change it to T0C + 1.35 if you want.

Mothblocks replied:
`T0C + 1.35` is better, though the best would be automatically deriving this. As in, dynamically getting the temperature of ice and water, and choosing the middle between that (and caching it).

SpookyTheFox replied:
Makes sense, but I'm not sure how to reference those values since they are private to those datums. I think. Do you have any examples elsewhere in the code I could look at?

SpookyTheFox replied:
I changed it to T0C + 1.35. Also did the emag module for consistency and to prevent identification in an edge case of fake beer mixed in a drink containing ice. I tested it again locally to make sure they both work.

SpookyTheFox committed:
Update borghydro.dm

Added comment line to var/dispensed_temperature as per documentation requirements

SpookyTheFox committed:
Update borghydro.dm

Changed 247.5K to T0C + 1.35. Also set reagent temperature for the hacked drink dispenser, to cover edge cases that might reveal it's use.

SpookyTheFox commented:
I've made the requested changes. In my second round of testing, I checked the normal soda and booze dispensers, they are set slightly too cold, and go the other way. Fixing that was not in the scope of this PR, which is focusing on borg shakers. I could add that though, it's a one line change to chem_dispenser.dm
